### PR TITLE
Convert `blog-with-comment` example to TypeScript

### PR DESCRIPTION
**Pull request by maxproske**
Converted `blog-with-comment` example to TypeScript to match Contribution docs.

- Updated all deps
- Fixed type errors (eg. textarea expects rows to be a number)

## Documentation / Examples

- [X] Make sure the linting passes by running `pnpm lint`
- [X] The examples guidelines are followed from [our contributing doc](https://github.com/vercel/next.js/blob/canary/contributing.md#adding-examples)

### CODE REVIEW

File: examples/blog-with-comment/package.json [hunk 1]
```diff
@@ -6,23 +6,27 @@
...
-    "autoprefixer": "^10.2.5",
-    "postcss": "^8.2.13",
-    "prettier": "^2.2.1",
-    "tailwindcss": "^2.1.2"
+    "@types/node": "^18.0.3",
+    "@types/react": "^18.0.15",
+    "@types/react-dom": "^18.0.6",
+    "autoprefixer": "^10.4.7",
+    "postcss": "^8.4.14",
+    "prettier": "^2.7.1",
```
Comment (balazsorban44) [inline examples/blog-with-comment/package.json:28]:
```suggestion
```

File: examples/blog-with-comment/pages/_app.tsx [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
 import Header from '../components/header'
 import { Auth0Provider } from '@auth0/auth0-react'
 
-export default function MyApp({ Component, pageProps }) {
+export default function MyApp({ Component, pageProps }: AppProps) {
   return (
     <Auth0Provider
-      clientId={process.env.NEXT_PUBLIC_AUTH0_CLIENT_ID}
-      domain={process.env.NEXT_PUBLIC_AUTH0_DOMAIN}
+      clientId={process.env.NEXT_PUBLIC_AUTH0_CLIENT_ID!}
```
Comment (balazsorban44) [inline examples/blog-with-comment/pages/_app.tsx:11]: I think in this case we could add something like an `environment.d.ts` file:

https://stackoverflow.com/a/53981706

What do you think?

File: examples/blog-with-comment/pages/_document.tsx [hunk 3]
```diff
@@ -1,8 +1,14 @@
-import NextDocument, { Html, Head, Main, NextScript } from 'next/document'
```
Comment (balazsorban44) [inline examples/blog-with-comment/pages/_document.tsx:1]: I think we can get rid of this file. `meta.title` is overridden in `_app` already to comply with https://github.com/vercel/next.js/blob/canary/errors/no-document-title.md

Let's remove it from this file and get rid of the `meta` object (just inline the `description` maybe)

I first thought we could remove the file entirely but there is some basic styling applied. Adding it back elsewhere would require creating a file in a different location, so there is no removed complexity.

We can turn this into a functional component though. See https://nextjs.org/docs/advanced-features/custom-document

**Comment by maxproske** [comment]
@balazsorban44 Thanks - Made all your requested changes!

**Pull request closed without merging**